tfe/resource_tfe_organization_token.go

```diff
@@ -0,0 +1,84 @@
+package tfe
+
+import (
+	"fmt"
+	"log"
+
+	tfe "github.com/hashicorp/go-tfe"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceTFEOrganizationToken() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceTFEOrganizationTokenCreate,
+		Read:   resourceTFEOrganizationTokenRead,
+		Delete: resourceTFEOrganizationTokenDelete,
+
+		Schema: map[string]*schema.Schema{
+			"organization": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"token": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceTFEOrganizationTokenCreate(d *schema.ResourceData, meta interface{}) error {
+	tfeClient := meta.(*tfe.Client)
+
+	// Get the organization name.
+	organization := d.Get("organization").(string)
+
+	log.Printf("[DEBUG] Create new token for organization: %s", organization)
+	token, err := tfeClient.OrganizationTokens.Generate(ctx, organization)
+	if err != nil {
+		return fmt.Errorf(
+			"Error creating new token for organization %s: %v", organization, err)
+	}
+
+	d.Set("token", token.Token)
```

Comment:
this should be below `d.SetId` and probably wants a comment explaining why it's here and not in the read (presumably that it's only returned once?)

Reply:
Why should it be below `d.SetId`?

Moved it and added a comment in the next commit.